From 0d6bc2738669d6e0aa94fd46cf1584f44ea7cf90 Mon Sep 17 00:00:00 2001
From: Dmitry Muhomor <muhomor.dmitry@gmail.com>
Date: Wed, 16 Oct 2024 21:41:18 +0300
Subject: [PATCH] ignore factory reset protection (FRP)

Pixel stock OS now enables FRP by default for everyone who has secure lock screen and a Google
account.
---
 .../android/server/pdb/PersistentDataBlockService.java   | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/services/core/java/com/android/server/pdb/PersistentDataBlockService.java b/services/core/java/com/android/server/pdb/PersistentDataBlockService.java
index fe9a85989cc9a..03f1d645b9192 100644
--- a/services/core/java/com/android/server/pdb/PersistentDataBlockService.java
+++ b/services/core/java/com/android/server/pdb/PersistentDataBlockService.java
@@ -608,6 +608,15 @@ void formatPartitionLocked(boolean setOemUnlockEnabled) {
     @VisibleForTesting
     boolean automaticallyDeactivateFrpIfPossible() {
         synchronized (mLock) {
+            if (hasFrpSecretMagic()
+                    // default secret is all zeros
+                    && !Arrays.equals(getFrpSecret(), new byte[FRP_SECRET_SIZE])) {
+                Slog.d(TAG, "deactivating FRP");
+                writeFrpMagicAndDefaultSecret();
+                Slog.d(TAG, "deactivated FRP");
+                return true;
+            }
+
             if (deactivateFrpWithFileSecret(mFrpSecretFile)) {
                 return true;
             }
